src/android/com/microsoftopentechnologies/intellij/helpers/azure/sdk/AzureSDKManagerImpl.java

```diff
@@ -1141,6 +1154,333 @@ public void downloadBlobFileContent(@NotNull StorageAccount storageAccount,
+
+                qmList.add(new QueueMessage(id, queueName, content, insertionTime, expirationTime, dequeueCount, subscriptionId));
+            }
+
+            return qmList;
+        } catch (Throwable t) {
+            throw new AzureCmdException("Error retrieving the Queue Message list", t);
+        }
+    }
+
+    @Override
+    public void clearQueue(@NotNull StorageAccount storageAccount, @NotNull Queue queue)
+            throws AzureCmdException {
+        try {
+            CloudQueueClient client = getCloudQueueClient(storageAccount);
+
+            CloudQueue cloudQueue = client.getQueueReference(queue.getName());
+            cloudQueue.clear();
+        } catch (Throwable t) {
+            throw new AzureCmdException("Error clearing the Queue", t);
+        }
+    }
+
+    @Override
+    public void createQueueMessage(@NotNull StorageAccount storageAccount,
+                                   @NotNull QueueMessage queueMessage,
+                                   int timeToLiveInSeconds)
+            throws AzureCmdException {
+        try {
+            CloudQueueClient client = getCloudQueueClient(storageAccount);
+
+            CloudQueue cloudQueue = client.getQueueReference(queueMessage.getQueueName());
+            cloudQueue.addMessage(new CloudQueueMessage(queueMessage.getContent()), timeToLiveInSeconds, 0, null, null);
+        } catch (Throwable t) {
+            throw new AzureCmdException("Error creating the Queue Message", t);
+        }
+    }
+
+    @NotNull
+    @Override
+    public QueueMessage dequeueFirstQueueMessage(@NotNull StorageAccount storageAccount, @NotNull Queue queue)
+            throws AzureCmdException {
+        try {
+            CloudQueueClient client = getCloudQueueClient(storageAccount);
+            String queueName = queue.getName();
+            String subscriptionId = storageAccount.getSubscriptionId();
+
+            CloudQueue cloudQueue = client.getQueueReference(queueName);
+            CloudQueueMessage cqm = cloudQueue.retrieveMessage();
+
+            String id = "";
+            String content = "";
+            Calendar insertionTime = new GregorianCalendar();
+            Calendar expirationTime = new GregorianCalendar();
+            int dequeueCount = 0;
+
+            if (cqm != null) {
+                id = Strings.nullToEmpty(cqm.getId());
+                content = Strings.nullToEmpty(cqm.getMessageContentAsString());
+
+                if (cqm.getInsertionTime() != null) {
+                    insertionTime.setTime(cqm.getInsertionTime());
+                }
+
+                if (cqm.getExpirationTime() != null) {
+                    expirationTime.setTime(cqm.getExpirationTime());
+                }
+
+                dequeueCount = cqm.getDequeueCount();
+
+                cloudQueue.deleteMessage(cqm);
+            }
+
+            return new QueueMessage(id, queueName, content, insertionTime, expirationTime, dequeueCount, subscriptionId);
```

Comment:
Should we instantiate the `QueueMessage` object *before* we call `deleteMessage`?